pwiz_tools/Skyline/Model/Results/OptStepChromatograms.cs

```diff
@@ -58,6 +58,31 @@ public static OptStepChromatograms FromChromatogram(ChromatogramInfo chromatogra
             return new OptStepChromatograms(chromatogramInfo.ProductMz, ImmutableList.Singleton(chromatogramInfo), 0);
         }
 
+        public static IEnumerable<OptStepChromatograms> FromChromatograms(IEnumerable<ChromatogramInfo> chromatogramInfos)
+        {
+            var curGroup = new List<ChromatogramInfo>();
+
+            OptStepChromatograms CurResult()
+            {
+                var result = new OptStepChromatograms(curGroup[curGroup.Count / 2].ProductMz, curGroup, (curGroup.Count - 1) / 2);
```

Comment:
Is it correct to be passing in "curGroup[curGroup.Count / 2].ProductMz"?
That means that you are assuming that you have an equal number of chromatogram groups above and below step 0.

With SRM optimization, that ends up not being the case if the collision energy is close to either the upper or lower limit of what the collision energy is allowed to be. Can the same sort of thing also happen with PRM optimization?

It might be that the optimization step integer should be added to ChromGroupHeaderInfo so that we can always tell which optimization step a chromatogram group is for.